### ISSUE TITLE
tf.contrib.ffmpeg.decode_video error

### ISSUE THREAD

**Issue by rongou** [tensorflow/tensorflow#20348]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Y
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 18.04
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: ('v1.8.0-4015-g44a1f241bc', '1.9.0-rc0')
- **Python version**: 2.7.15rc1
- **Bazel version (if compiling from source)**: 0.15.0
- **GCC/Compiler version (if compiling from source)**: 6.4.0
- **CUDA/cuDNN version**: 9.0.176/7.0.5
- **GPU model and memory**: GeForce GTX 1080 Ti (11GB)
- **Exact command to reproduce**:
```python
import tensorflow as tf
with tf.Session() as sess:
    movie_bin = tf.read_file('5205acf1-3d30b48d.mov')
    movie = tf.contrib.ffmpeg.decode_video(movie_bin)
    movie_ev = movie.eval()
    print("****", len(movie_ev))
```
### Describe the problem

The `tf.contrib.ffmpeg.decode_video` op throws an error and causes a core dump.

My ffmpeg version (`apt get install ffmpeg` on ubuntu 18.04):
```shell
ffmpeg version 3.4.2-2 Copyright (c) 2000-2018 the FFmpeg developers
  built with gcc 7 (Ubuntu 7.3.0-16ubuntu2)
  configuration: --prefix=/usr --extra-version=2 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --enable-gpl --disable-stripping --enable-avresample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librubberband --enable-librsvg --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-omx --enable-openal --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libopencv --enable-libx264 --enable-shared
  libavutil      55. 78.100 / 55. 78.100
  libavcodec     57.107.100 / 57.107.100
  libavformat    57. 83.100 / 57. 83.100
  libavdevice    57. 10.100 / 57. 10.100
  libavfilter     6.107.100 /  6.107.100
  libavresample   3.  7.  0 /  3.  7.  0
  libswscale      4.  8.100 /  4.  8.100
  libswresample   2.  9.100 /  2.  9.100
  libpostproc    54.  7.100 / 54.  7.100
```
### Source code / logs
```shell
2018-06-27 09:15:22.489692: F tensorflow/contrib/ffmpeg/default/ffmpeg_lib.cc:405] Non-OK-status: ReadInfoFile(stderr_filename, width, height, frames) status: Unknown: Not enough video info returned by FFmpeg [0, 720, 1280, 3]Could not read FFmpeg stderr file: /tmp/tmp_file_tensorflow_3_ILW8Ht.err
Aborted (core dumped)
```

**Comment by rongou** [comment]
May be similar to #17533. @yongtang

**Comment by rongou** [comment]
Here is the err file:
```shell
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/tmp/tmp_file_tensorflow_1_RNEIEt':
  Metadata:
    major_brand     : qt  
    minor_version   : 512
    compatible_brands: qt  
    encoder         : Lavf57.71.100
  Duration: 00:00:40.06, start: 0.000000, bitrate: 4036 kb/s
    Stream #0:0(eng): Video: h264 (High) (avc1 / 0x31637661), yuv420p(tv, smpte170m/bt709/bt709), 720x1280, 4034 kb/s, 30.16 fps, 59.94 tbr, 19200 tbn, 38400 tbc (default)
    Metadata:
      rotate          : 90
      handler_name    : DataHandler
      encoder         : H.264
    Side data:
      displaymatrix: rotation of -90.00 degrees
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> rawvideo (native))
Output #0, image2pipe, to '/tmp/tmp_file_tensorflow_2_q9AqGY.raw':
  Metadata:
    major_brand     : qt  
    minor_version   : 512
    compatible_brands: qt  
    encoder         : Lavf57.83.100
    Stream #0:0(eng): Video: rawvideo (RGB[24] / 0x18424752), rgb24, 1280x720, q=2-31, 1325778 kb/s, 59.94 fps, 59.94 tbn, 59.94 tbc (default)
    Metadata:
      encoder         : Lavc57.107.100 rawvideo
      handler_name    : DataHandler
    Side data:
      displaymatrix: rotation of -0.00 degrees
More than 1000 frames duplicated 5370300kB time=00:00:33.18 bitrate=1325778.2kbits/s dup=992 drop=5 speed=6.61x    
frame= 2401 fps=352 q=-0.0 Lsize= 6482700kB time=00:00:40.05 bitrate=1325778.2kbits/s dup=1198 drop=5 speed=5.88x    
video:6482700kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.000000%
```

**Comment by yongtang** [comment]
@rongou Thanks for the report, I will take a look.

**Comment by yongtang** [comment]
Created a PR #20366 for the fix.

**Comment by rongou** [comment]
@yongtang sent you an alternative fix: #20388. Tested it on 70 .mov files from the [BDD100K dataset](http://bdd-data.berkeley.edu/).

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rongou** [comment]
ping @yifeif

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rongou: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rongou: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rongou: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rongou: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @rongou, @yifeif: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by piercus** [comment]
I was blocked by this also on my side.
I'm sharing a dirty workaround here (might be useful to someone else), that is "patching" ffmpeg (no need to recompile tensorflow).

**Warning** this is dirty and this may broke your ffmpeg install so you should revert the patch after you did it

### Save ffmpeg binary as `ffmpeg-unpatched`
```
ffmpeg_path=$(which ffmpeg)
mv $ffmpeg_path ${ffmpeg_path}-unpatched
```

### Create a patched-ffmpeg script

save a file named `ffmpeg` somewhere in your path
```
#!/bin/bash

# create a temporary file to put ffmpeg input
rand=$(shuf -i1-100000 -n1)
filename=/tmp/$rand.ffmpegout

# call ffmpeg and redirect stderr to temporary file
ffmpeg-unpatched "$@" > /dev/null 2> $filename

echo "This ffmpeg is a bash script calling ffmpeg in order to patch it for tensorflow/issues/20348"

# output ffmpeg results in an order that is matching tensorflow expectations
cat $filename | grep -A 100 "Output" >&2
cat $filename | grep -A 1 "Stream mapping" >&2
cat $filename | grep "frame=" >&2

# Uncomment this line for bugfixing (to visualize your frames numbers line)
# cat $filename | grep "frame="

rm -f $filename

exit
```

### Run tensorflow again

now it should work 

### Revert
```
mv $ffmpeg_path-unpatched ${ffmpeg_path}
```

**Comment by tensorflowbutler** [comment]
Nagging Assignees @rongou, @yifeif: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @rongou, @yifeif: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by ymodak** [comment]
`tf.contrib` module is no longer supported.
Please test with `TF_IO` module https://www.tensorflow.org/io/api_docs/python/tfio/experimental/ffmpeg/decode_video
and raise a new issue if problem still persists. Thanks!
